File: AdventOfCode/2022/01/calories.py
#input = numebr of calories in a snack, empty line separates elves 

with open("AdventOfCode\\2022\\01\\input.txt") as f:
    elves_calories = []
    carry = []
    for line in f:
        line = line.strip()
        if line == "":
            elves_calories.append(sum(carry))
            carry = []
            continue
        carry.append(int(line))
elves_calories = sorted(elves_calories, reverse=True)
print(elves_calories[0])
print(elves_calories[0] + elves_calories[1] + elves_calories[2])

File: AdventOfCode/2022/01/calories.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector> //vector - dynamic array
#include <bits/stdc++.h> //sort

using namespace std;

int main(){
    //reading file 
    fstream file;
    file.open("C:\\Users\\matej\\Documents\\Projects\\AdventOfCode\\2022\\01\\input.txt", ios::in);
    
    if (file.is_open()){
        string line;
        vector<int> elves_calories;
        int total = 0;

        //getting the total callories per elf
        
        while (getline(file,line)){
            if (line != ""){
                total += (stoi(line));
            }else{
                elves_calories.push_back(total);
                total = 0;
            }
        }

        //sorting the total callories per elf
        sort(elves_calories.begin(), elves_calories.end(), greater<int>());
        int first = 0; int second = 0; int third = 0;
        
        //printing answers for part1 and part2
        cout << elves_calories[0] << endl;
        cout << elves_calories[0] + elves_calories[1] + elves_calories[2];

    //error testing for presious stupid mistakes
    }else if (!file.is_open()){
        cout << "Unable to open file";
    }
    file.close();

    return 0;
}
